tests/juniperQfxCopper/juniper_qfx_copper_protocol_test.py

```diff
@@ -0,0 +1,1049 @@
+
+        conf = result._NCElement__result.xml
+        assert_that(conf, contains_regex(
+                '<configuration xmlns="http://xml.juniper.net/xnm/1.1/xnm" junos:commit-localtime="[^"]*" junos:commit-seconds="[^"]*" junos:commit-user="[^"]*">'))
+
+        assert_that(result.xpath("data/configuration/interfaces/interface/unit/family/ethernet-switching"),
+                    has_length(4))
+        assert_that(result.xpath("data/configuration/vlans/vlan"), has_length(0))
+
+    def test_lock_edit_candidate_add_vlan_and_commit(self):
+        with self.nc.locked(target='candidate'):
+            result = self.nc.edit_config(target='candidate', config=dict_2_etree({
+                "config": {
+                    "configuration": {
+                        "vlans": {
+                            "vlan": {
+                                "name": "VLAN2999",
+                            }
+                        }
+                    }
+                }}))
+            assert_that(result.xpath("//rpc-reply/ok"), has_length(1))
+
+            result = self.nc.commit()
+            assert_that(result.xpath("//rpc-reply/ok"), has_length(1))
+
+        result = self.nc.get_config(source="running")
+
+        assert_that(result.xpath("data/configuration/vlans/vlan"), has_length(1))
+
+        self.edit({
+            "vlans": {
+                "vlan": {
+                    XML_ATTRIBUTES: {"operation": "delete"},
+                    "name": "VLAN2999"
+                }
+            }
+        })
+
+        self.nc.commit()
+
+        result = self.nc.get_config(source="running")
+        assert_that(result.xpath("data/configuration/vlans/vlan"), has_length(0))
+
+    def test_locking_fails_if_changes_are_being_made(self):
+        nc2 = self.create_client()
+
+        try:
+            self.nc.edit_config(target='candidate', config=dict_2_etree({
+                "config": {
+                    "configuration": {
+                        "vlans": {
+                            "vlan": [
+                                {"name": "VLAN2999"},
+                                {"description": "WHAAT"}
+                            ]
+                        }
+                    }
+                }}))
+
+            with self.assertRaises(RPCError):
+                with nc2.locked(target='candidate'):
+                    self.fail('Should not be able to lock an edited configuration')
+
+        finally:
+            self.nc.discard_changes()
+            nc2.close_session()
+
+    def test_double_locking_with_two_sessions(self):
+        nc2 = self.create_client()
+
+        try:
+            with self.nc.locked(target='candidate'):
+                with self.assertRaises(RPCError):
+                    with nc2.locked(target='candidate'):
+                        self.fail("The second lock should not have worked.")
+
+        finally:
+            nc2.close_session()
+
+    def test_bad_configuration_element(self):
```

Comment:
Is this the same test as what we had in the base juniper? I would rather see this test only the specifics. Is this doable?